blocks/section.ts

```diff
@@ -1,18 +1,166 @@
-import { newComponentBlock } from "$live/blocks/utils.ts";
+// deno-lint-ignore-file no-explicit-any
+import { HandlerContext } from "$fresh/server.ts";
+import { HttpContext, LiveConfig } from "$live/blocks/handler.ts";
 import StubSection from "$live/components/StubSection.tsx";
-import { InstanceOf } from "$live/engine/block.ts";
+import {
+  Block,
+  BlockModule,
+  BlockModuleRef,
+  ComponentFunc,
+  InstanceOf,
+  PreactComponent,
+} from "$live/engine/block.ts";
+import { BaseContext, Resolver } from "$live/engine/core/resolver.ts";
+import { PromiseOrValue, UnPromisify } from "$live/engine/core/utils.ts";
+import { introspectAddr } from "$live/engine/introspect.ts";
+import { Schemeable, TransformContext } from "$live/engine/schema/transform.ts";
+import { LoaderContext } from "$live/types.ts";
+import { omit } from "$live/utils/object.ts";
+import { DocNode } from "https://deno.land/x/deno_doc@0.58.0/lib/types.d.ts";
+import { JSX } from "preact";
 
 export type Section = InstanceOf<typeof sectionBlock, "#/root/sections">;
 
-const sectionBlock = newComponentBlock("sections", (_, ctx) => {
+export interface SectionModule<TConfig = any, TProps = any> extends
+  BlockModule<
+    ComponentFunc<TProps>,
+    JSX.Element | null,
+    PreactComponent
+  > {
+  loadData?: (
+    req: Request,
+    ctx: LoaderContext<LiveConfig<TConfig, any>>,
+  ) => PromiseOrValue<Partial<TProps>>;
+}
+
+const omitIfObj = (schemeable: Schemeable, omitKeys: string[]): Schemeable => {
+  if (schemeable.type !== "object" || omitKeys.length === 0) {
+    return schemeable;
+  }
   return {
-    Component: StubSection,
-    props: {
-      component: ctx.resolveChain[ctx.resolveChain.length - 1],
-    },
+    ...schemeable,
+    name: schemeable.name ? `omit${schemeable.name}` : undefined,
+    value: omit(schemeable.value, ...omitKeys),
   };
+};
+
+const componentWith = (
+  resolver: string,
+  componentFunc: ComponentFunc,
+) =>
+<TProps = any>(props: TProps, resolveChain: string[]) => ({
+  Component: componentFunc,
+  props,
+  metadata: {
+    component: resolver,
+    resolveChain,
+    id: resolveChain.length > 0 ? resolveChain[0] : undefined,
+  },
 });
 
+const sectionBlock: Block<SectionModule> = {
+  type: "sections",
+  introspect: async (
```

Comment:
what this does is:

Get the props from the default function and the loadData function.

Create a JSONSchema which is the intersection of both props except all keys from the loadData return